### PR TITLE
feat(bootstrap-4): Added and Modified Widgets, Refactored Form Component

### LINKED ISSUE #14: [Bootstrap 4]: Create PasswordWidget
### Prerequisites

- [x] I have read the [documentation](https://react-jsonschema-form.readthedocs.io/);
- [x] In the case of a bug report, I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [x] Ideally, I'm providing a [sample JSFiddle](https://jsfiddle.net/n1k0/f2y3fq7L/6/) or a [shared playground link](https://rjsf-team.github.io/react-jsonschema-form/) demonstrating the issue.

### Description

Creating the PasswordWidget according to the Bootstrap 4 semantics

### Steps to Reproduce

1. Load bootstrap-4 theme in the playground

### PR DESCRIPTION
**Pull request by Xtremilicious**
### Reasons for making this change

Added Password, UpDown(Number), Email, URL, Select, FieldTemplate and ErrorList widgets, styled according to Bootstrap 4 semantics. Added support for invalid/error input boxes. Modified Text widget to support `schema.examples`, as a `<datalist>` component. Refactored `Form` component to remove `MuiForm` naming.

fixed #14, #15, #34, #8, #23, #19, #24  

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [x] **I'm adding or updating code**
  - [ ] I've added and/or updated tests
  - [ ] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
* [x] **I'm adding a new feature**
  - [x] I've updated the playground with an example use of the feature

### CODE REVIEW

**Comment by Xtremilicious** [comment]
@jevakallio Kindly review the PR, so that the PR doesn't become huge

File: packages/bootstrap-4/src/EmailWidget/EmailWidget.tsx [hunk 1]
```diff
@@ -2,7 +2,7 @@ import React from "react";
 import TextWidget, { TextWidgetProps } from "../TextWidget";
 
 const EmailWidget = (props: TextWidgetProps) => {
-  return <TextWidget type="email" {...props} />;
+  return <TextWidget {...props} type="email" />;
```
Comment (anikethsaha) [inline packages/bootstrap-4/src/EmailWidget/EmailWidget.tsx:5]: @Xtremilicious I guess for this to work lint free, I think `type` has to be optional in `index.d.ts` in `widgetProps` .

may be , you can check this

Comment (jevakallio) [reply packages/bootstrap-4/src/EmailWidget/EmailWidget.tsx:5]: Looks like the `type` prop type is being removed. Once that lands, we can sync and merge that change back. This change can then be reverted (or kept, it doesn't matter as the type prop should not be overrideble)

https://github.com/rjsf-team/react-jsonschema-form/pull/1826

Comment (Xtremilicious) [reply packages/bootstrap-4/src/EmailWidget/EmailWidget.tsx:5]: @anikethsaha should we merge this or wait for now?

Comment (anikethsaha) [reply packages/bootstrap-4/src/EmailWidget/EmailWidget.tsx:5]: we have to wait for rjsf-team#1826

**Comment by anikethsaha** [comment]
@Xtremilicious resolve the merge conflict

File: packages/bootstrap-4/src/EmailWidget/EmailWidget.tsx (hunk 1 above)
Comment (Xtremilicious) [reply packages/bootstrap-4/src/EmailWidget/EmailWidget.tsx:5]: @anikethsaha I resolved the conflicts, you can now merge 👍

**Comment by Xtremilicious** [comment]
> @Xtremilicious resolve the merge conflict

@anikethsaha I resolved the conflicts, you can now merge 👍

**Comment by anikethsaha** [comment]
pending review from @jevakallio

File: packages/bootstrap-4/src/FieldTemplate/FieldTemplate.tsx [hunk 2]
```diff
@@ -2,48 +2,46 @@ import React from "react";
...
       ) : null}
       {rawErrors.length > 0 && (
-        <List dense={true} disablePadding={true}>
+        <ListGroup as="ul">
           {rawErrors.map((error, i: number) => {
             return (
-              <ListItem key={i} disableGutters={true}>
-                <FormHelperText id={id}>{error}</FormHelperText>
-              </ListItem>
+              <ListGroup.Item as="li" key={i} variant="danger">
```
Comment (jevakallio) [inline packages/bootstrap-4/src/FieldTemplate/FieldTemplate.tsx:28]: We shouldn't use indexes as keys. Here's why: https://medium.com/@vraa/why-using-an-index-as-key-in-react-is-probably-a-bad-idea-7543de68b17c

I'd rather use `error` as key in this case, assuming it's a plain string?

File: packages/bootstrap-4/src/TextWidget/TextWidget.tsx [hunk 3]
```diff
@@ -36,23 +36,33 @@ const TextWidget = ({
...
         type={type || (schema.type as string)}
         value={value || value === 0 ? value : ""}
         onChange={_onChange}
         onBlur={_onBlur}
         onFocus={_onFocus}
         {...textFieldProps}
       />
+      {schema.examples ? (
+        <datalist id={`examples_${id}`}>
+          {(schema.examples as any)
```
Comment (jevakallio) [inline packages/bootstrap-4/src/TextWidget/TextWidget.tsx:59]: Can we avoid `as any` by improving the type definition? If not, this is fine, but not ideal.

**Comment by Xtremilicious** [comment]
@anikethsaha Merging after code fixes. Kindly pull this version.

**Pull request merged**